CMakeLists.txt

```diff
@@ -133,15 +133,15 @@ elseif (${Z3_FOUND})
   message("Z3 SMT solver found. This enables optional SMT checking with Z3.")
 endif()
 
-find_package(CVC4 QUIET)
-if (${CVC4_FOUND})
-  add_definitions(-DHAVE_CVC4)
-  message("CVC4 SMT solver found. This enables optional SMT checking with CVC4.")
+find_program(CVC5_PATH cvc5)
+if (CVC5_PATH)
+  add_compile_definitions(CVC5_PATH=${CVC5_PATH})
+  message("cvc5 SMT solver found. This enables optional SMT checking with cvc5.")
 endif()
```

Comment:
I was about to say that this is bad, since if you then use ``CVC5_PATH`` somewhere, it'll get the path hardcoded - but you actually don't use it anywhere, right :-)? So the ``add_compile_definitions`` is superfluous anyways?

Should we even have any kind of ``cmake`` logic here? I've not looked at the rest of the PR yet, but I'd guess that CVC5 is purely a runtime dependency with the change, so we don't need to do anything special at cmake configure time, do we?

Reply:
That's a good point. I thought we need to give some sort of a message at configuration time, but most likely we don't.
Yes, `cvc5` is now a runtime dependency. It works the same way as `Eldarica`, so basically the executable must be present in PATH.

I'll remove this `cmake` logic.

Reply:
I am wondering about the message below, which warns if `Z3` nor `CVC4` (now `cvc5`) is not enabled/has not been found.

Should there be any information/warning at cmake configure time about available solvers?